objects/src/transaction/tx_args.rs

```diff
@@ -37,32 +37,33 @@ impl TransactionArgs {
     /// arguments.
     ///
     /// If tx_script is provided, this also adds all mappings from the transaction script inputs
-    /// to the advice map.
+    /// to the advice inputs' map.
     pub fn new(
         tx_script: Option<TransactionScript>,
         note_args: Option<BTreeMap<NoteId, Word>>,
-        mut advice_map: AdviceMap,
+        mut advice_inputs: AdviceInputs,
     ) -> Self {
```

Comment:
I would maybe keep the previous interface here and pass only the advice map. Then, this map could be added to the inputs internally.

If we do want to pass in the full `advice_inputs` into the constructor, we need to make sure that `advice_inputs.stack` is empty. But I think this may complicate things a bit too much.